DOCS: Document the changes for EQL in Elasticsearch (#39)
endgameinc/eql: docs/index.rst

EQL also has a preprocessor that can perform parse and translation time evaluation, allowing for easily sharable components between queries.

.. note::
   This documentation is about EQL for Elastic Endgame. Several syntax changes were made to `bring Event Query Language to the Elastic Stack <https://www.elastic.co/guide/en/elasticsearch/reference/current/eql.html>`_:
   

>    - All operators and functions are case-sensitive. For example, ``process_name == "cmd.exe"`` is different from ``process_name == "Cmd.exe"``.

Reviewer: Small nit.
```suggestion
   - Most operators and functions are now case-sensitive. For example, ``process_name == "cmd.exe"`` is no longer equivalent to ``process_name == "Cmd.exe"``.
```

>    - Use ``:`` to perform case-insensitive equals. For example, ``process_name : "cmd.exe"`` is identical to ``process_name : "Cmd.exe"``.

Reviewer: I think the original is probably clear, but I'd call this out as an operator.
```suggestion
   - For case-insensitive equality comparisons, use the ``:`` operator. For example, ``process_name : "cmd.exe"`` is equivalent to ``process_name : "Cmd.exe"``.
```

   - The ``==`` and ``!=`` operators no longer expand wildcard characters. ``process_name == "cmd*.exe"`` will interpret ``*`` as a literal asterisk, not a wildcard character.

>    - ``==`` can no longer be substitutde for ``=``

Reviewer: Minor typo:
```suggestion
   - ``=`` can no longer be substituted for the ``==`` operator.
```

>    - ``?"`` and ``?'`` are not used for raw strings. Instead, use the new ``"""..."""`` syntax

Reviewer: ```suggestion
   - ``?"`` and ``?'`` no longer indicate raw strings. Use the ``"""..."""`` syntax instead.
```
